Sources/PackageGraph/RepositoryPackageContainer.swift

```diff
@@ -266,24 +270,26 @@ public class RepositoryPackageContainer: PackageContainer, CustomStringConvertib
     }
    
     private func loadManifest(at revision: Revision, version: Version?) throws -> Manifest {
-        let fs = try repository.openFileView(revision: revision)
-        let packageURL = identifier.repository.url
-
-        // Load the tools version.
-        let toolsVersion = try toolsVersionLoader.load(at: .root, fileSystem: fs)
-
-        // Validate the tools version.
-        try toolsVersion.validateToolsVersion(
-            self.currentToolsVersion, version: revision.identifier, packagePath: packageURL)
-
-        // Load the manifest.
-        return try manifestLoader.load(
-            package: AbsolutePath.root,
-            baseURL: packageURL,
-            version: version,
-            toolsVersion: toolsVersion,
-            packageKind: identifier.kind,
-            fileSystem: fs)
+        try self.manifestsCache.memoize(revision) {
```

Comment:
It looks as if this will avoid memoizing a failure.  Is that a correct understanding (to be clear, I think that's the right behavior).

Reply:
yes, I prefer not memoize failures